From 7ad61a386bb804ae562966ae23baf5f4deee014b Mon Sep 17 00:00:00 2001
From: Aishwarya Pant <aishpant@gmail.com>
Date: Mon, 10 Apr 2017 19:13:50 +0530
Subject: [PATCH 27/37] staging: rtl8723bs: remove null test before kfree

kfree(..) on a NULL pointer is a no-op; the null test here is redundant.
Detected by coccicheck.

Signed-off-by: Aishwarya Pant <aishpant@gmail.com>
Acked-by: Larry Finger <Larry.Finger@lwfinger.net>
Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---
 drivers/staging/rtl8723bs/hal/rtl8723b_hal_init.c |  6 ++----
 drivers/staging/rtl8723bs/hal/sdio_ops.c          |  6 ++----
 drivers/staging/rtl8723bs/os_dep/ioctl_cfg80211.c |  8 ++-----
 drivers/staging/rtl8723bs/os_dep/ioctl_linux.c    | 26 ++++++-----------------
 drivers/staging/rtl8723bs/os_dep/xmit_linux.c     |  6 ++----
 5 files changed, 15 insertions(+), 37 deletions(-)

--- a/drivers/staging/rtl8723bs/hal/rtl8723b_hal_init.c
+++ b/drivers/staging/rtl8723bs/hal/rtl8723b_hal_init.c
@@ -995,8 +995,7 @@ if (1) {
 		rtw_hal_set_hwreg(padapter, HW_VAR_EFUSE_USAGE, (u8 *)&efuse_usage);
 	}
 
-	if (efuseTbl)
-		kfree(efuseTbl);
+	kfree(efuseTbl);
 }
 
 static void hal_ReadEFuse_BT(
@@ -1128,8 +1127,7 @@ static void hal_ReadEFuse_BT(
 	}
 
 exit:
-	if (efuseTbl)
-		kfree(efuseTbl);
+	kfree(efuseTbl);
 }
 
 static void Hal_ReadEFuse(
--- a/drivers/staging/rtl8723bs/hal/sdio_ops.c
+++ b/drivers/staging/rtl8723bs/hal/sdio_ops.c
@@ -602,8 +602,7 @@ static s32 _sdio_local_read(
 	if (!err)
 		memcpy(pbuf, ptmpbuf, cnt);
 
-	if (ptmpbuf)
-		kfree(ptmpbuf);
+	kfree(ptmpbuf);
 
 	return err;
 }
@@ -646,8 +645,7 @@ s32 sdio_local_read(
 	if (!err)
 		memcpy(pbuf, ptmpbuf, cnt);
 
-	if (ptmpbuf)
-		kfree(ptmpbuf);
+	kfree(ptmpbuf);
 
 	return err;
 }
--- a/drivers/staging/rtl8723bs/os_dep/ioctl_cfg80211.c
+++ b/drivers/staging/rtl8723bs/os_dep/ioctl_cfg80211.c
@@ -1170,10 +1170,7 @@ static int cfg80211_rtw_add_key(struct w
 	}
 
 addkey_end:
-	if (param)
-	{
-		kfree((u8 *)param);
-	}
+	kfree((u8 *)param);
 
 	return ret;
 
@@ -1968,8 +1965,7 @@ static int rtw_cfg80211_set_wpa_ie(struc
 		pairwise_cipher, padapter->securitypriv.ndisencryptstatus, padapter->securitypriv.ndisauthtype));
 
 exit:
-	if (buf)
-		kfree(buf);
+	kfree(buf);
 	if (ret)
 		_clr_fwstate_(&padapter->mlmepriv, WIFI_UNDER_WPS);
 	return ret;
--- a/drivers/staging/rtl8723bs/os_dep/ioctl_linux.c
+++ b/drivers/staging/rtl8723bs/os_dep/ioctl_linux.c
@@ -766,9 +766,7 @@ static int wpa_set_encryption(struct net
 
 exit:
 
-	if (pwep) {
-		kfree((u8 *)pwep);
-	}
+	kfree((u8 *)pwep);
 	return ret;
 }
 
@@ -926,7 +924,7 @@ static int rtw_set_wpa_ie(struct adapter
 
 exit:
 
-	if (buf) kfree(buf);
+	kfree(buf);
 
 	return ret;
 }
@@ -2502,10 +2500,7 @@ static int rtw_wx_set_enc_ext(struct net
 	ret =  wpa_set_encryption(dev, param, param_len);
 
 exit:
-	if (param)
-	{
-		kfree((u8 *)param);
-	}
+	kfree((u8 *)param);
 
 	return ret;
 }
@@ -4135,11 +4130,7 @@ static int rtw_set_encryption(struct net
 	}
 
 exit:
-
-	if (pwep)
-	{
-		kfree((u8 *)pwep);
-	}
+	kfree((u8 *)pwep);
 
 	return ret;
 
@@ -5785,12 +5776,9 @@ static int rtw_ioctl_wext_private(struct
 	}
 
 exit:
-	if (input)
-		kfree(input);
-	if (buffer)
-		kfree(buffer);
-	if (output)
-		kfree(output);
+	kfree(input);
+	kfree(buffer);
+	kfree(output);
 
 	return err;
 }
--- a/drivers/staging/rtl8723bs/os_dep/xmit_linux.c
+++ b/drivers/staging/rtl8723bs/os_dep/xmit_linux.c
@@ -76,10 +76,8 @@ int rtw_os_xmit_resource_alloc(struct ad
 
 void rtw_os_xmit_resource_free(struct adapter *padapter, struct xmit_buf *pxmitbuf, u32 free_sz, u8 flag)
 {
-	if (free_sz > 0) {
-		if (pxmitbuf->pallocated_buf)
-			kfree(pxmitbuf->pallocated_buf);
-	}
+	if (free_sz > 0)
+		kfree(pxmitbuf->pallocated_buf);
 }
 
 #define WMM_XMIT_THRESHOLD	(NR_XMITFRAME*2/5)
